gnovm/pkg/gnolang/preprocess.go

```diff
@@ -2540,6 +2354,219 @@ func preprocess1(store Store, ctx BlockNode, n Node) Node {
 	return nn
 }
 
+// defineOrDecl merges the code logic from op define (:=) and declare (var/const).
+func defineOrDecl(
+	store Store,
+	bn BlockNode,
+	isConst bool,
+	nameExprs []NameExpr,
+	typeExpr Expr,
+	valueExprs []Expr,
+) {
+	numNames := len(nameExprs)
+	numVals := len(valueExprs)
+
+	if numVals > 1 && numNames != numVals {
+		panic(fmt.Sprintf("assignment mismatch: %d variable(s) but %d value(s)", numNames, numVals))
+	}
+
+	if numNames < 1 {
+		panic("must have at least one name to assign")
+	}
+
+	sts := make([]Type, numNames) // static types
+	tvs := make([]TypedValue, numNames)
+
+	if numNames > 1 && len(valueExprs) == 1 {
+		parseMultipleAssignFromOneExpr(sts, tvs, store, bn, nameExprs, typeExpr, valueExprs[0])
+	} else {
+		parseAssignFromExprList(sts, tvs, store, bn, isConst, nameExprs, typeExpr, valueExprs)
+	}
+
+	node := skipFile(bn)
+
+	for i := 0; i < len(sts); i++ {
+		nx := nameExprs[i]
+		if nx.Name == blankIdentifier {
+			nx.Path = NewValuePathBlock(0, 0, blankIdentifier)
+		} else {
+			node.Define2(isConst, nx.Name, sts[i], tvs[i])
+			nx.Path = bn.GetPathForName(nil, nx.Name)
+		}
+	}
+}
+
+// parseAssignFromExprList parses assignment to multiple variables from a list of expressions.
+func parseAssignFromExprList(
+	sts []Type,
+	tvs []TypedValue,
+	store Store,
```

Comment:
this might be an OCD request, but I suggest the parameter order to be (store, bn, others...)  to maintain consistency.

Reply:
In fact, in this case, the function will modify the given sts and tvs so I'd like to put them in first place. For the rest of args I'm trying to keep the consistency with other functions args :)